### DISCUSSION TITLE
Ajuda para implementar o metodo Put, no curso 1974

### DISCUSSION THREAD

**Discussion by RafaelDifini** [balta-io/forum#280] [Tira Dúvidas] [answered]
Boa tarde, Balta, to aqui quebrando a cabeça a dois dias tentando implementar o método put que nao foi implementado durante o curso, ate consegui com o delete eu acho, porem o put, nao entendi mesmo, vou te mostrar como ta atualmente o rascunho do metodo.
![image](https://user-images.githubusercontent.com/79666617/222928198-769adda1-eebc-4840-a8f4-b5c8ad1925e7.png)

o comando Update, no repositorio ( NÃO SEI SE A QUERY TA CERTA)
![image](https://user-images.githubusercontent.com/79666617/222928241-cd18c2c0-db0e-4edb-a7c7-98e1403b8de4.png)
o command
![image](https://user-images.githubusercontent.com/79666617/222926921-e2402ef1-23c3-43e8-a52f-cae3c9adb109.png)
e o command result:
![image](https://user-images.githubusercontent.com/79666617/222926930-bd09e7d1-d6bf-4c5c-bc15-36a82196d3ec.png)

por favor balta, da essa mão ai, ja fiz uns 3 cursos que voce aprensentou CQRS, esse eu foi oque eu mais entendi, porem pra implementar esse put ficou algo muito complicado pra mim, nao sei se tenho que fazer por Guid, por Customer, etc.

**Comment by andrebaltieri** [comment, marked as answer]
Boa noite, @RafaelDifini, claro, como posso ajudar?

Sobre a query, o ideal é sempre rodar ela no SQL Server antes, de resto me pareceu tudo OK, deu algum erro?

**Reply by RafaelDifini** [reply]
Boa noite Balta, Infelizmente minha implementação não deu certo, acabei me confundindo muito pra implementar com CQRS, e toda essa coisa de commands, commandResults, Queries etc, sei que é muito corrido seu dia e esse curso é legado, mas se voce puder um dia implementar esse put no seu repositorio, seria mt bacana, esclarecer todas essas duvidas. obrigado.

**Reply by andrebaltieri** [reply]
Posso, claro... mas olha... CQRS é realmente complicado... minha sugestão é revisar os cursos anteriores, colocar mais a mão na massa com OOP, treinar mais, ai quando chegar aqui, vai estar mais fácil.

Eu mesmo patinei muito com CQRS e DDD, nos cursos parece tudo tranquilo, mas foram anos de estudo sobre o assunto... que inclusive continuam me atormentando :D
